torch/distributed/distributed_c10d.py

```diff
@@ -245,6 +246,36 @@ def _check_tensor_list(param, param_name):
                            "to be of type List[torch.Tensor].".format(param_name))
 
 
+def _check_op_list(op_list, list_name):
+    """
+    Helper to check that the ``op_list`` is a list of functions.
+
+    """
+    if not isinstance(op_list, list) or \
+       not all(op in [isend, irecv] for op in op_list):
+        raise RuntimeError(f"Invalid function. Expected fucntion in `{op_list}` "
+                           f"to be of type torch.distributed.isend or"
+                           f"torch.distributed.irecv.")
+
+
+def _check_input_length(op_list, tensor_list, peer_list, group_list, tag_list):
+    """
+    Helper to check that all the inputs have the same length.
+
+    """
+    expected_length = len(op_list)
+    check_list = [tensor_list, peer_list]
+    if group_list is not None:
+        check_list.append(group_list)
```

Comment:
Didn't quite get why we enforce group_list to be the same length with op_list. I guess a common use case is to have a local process group and issue batched isend/irecv through this single process group? And multiple NCCL communicators (process groups) of a process is not safe according to Nvidia

Reply:
I want to keep the interface same as isend/irecv to provide flexibility for users. With the current way, it can cover cases 1)all the isend/irecv are in a single process group, users can create a list and pass it to group_list. 2) different ops are from different groups. I agree it is not safe for NCCL but this batch interface could potentially be used for other backends and not just GPUs.